File: COTE/LeetCode/easy/palindrome-number.js
/**
 * @param {number} x
 * @return {boolean}
 */
 var isPalindrome = function(x) {
   let strNum = x.toString()
   let strLeng = strNum.length
   let right;
   if(strNum.slice(0,1)=='-1')return false

   let midNum = Math.floor(strLeng/2)
   let left = strNum.slice(0,midNum)

   if(strLeng%2==0){
       right=strNum.slice(midNum, strLeng)
   } else {
       right=strNum.slice(midNum+1, strLeng)
   }

   let reversedRight = reverse(right)

   if(left==reversedRight){
       return true
   } else {
       return false
   }
}

function reverse(s){
console.log(s.split(""))
return s.split("").reverse().join("");
}

File: COTE/LeetCode/easy/palindrome-number.py
class Solution:
    def isPalindrome(self, x: int) -> bool:
        strNum = str(x)
        right=''
        if strNum[0]=='-':
            return False
        
        midNum =len(strNum)//2
        left= strNum[0:midNum]
        if len(strNum)%2==0:
            right= strNum[midNum:len(strNum)]
        else:
            right= strNum[midNum+1:len(strNum)]
            
        reversedRight=reverseString(right)
        if left==reversedRight:
            return True
        else:
            return False
        
def reverseString(x):
        return x[::-1]